src/bref/src/BrefRunner.php

```diff
@@ -31,7 +31,15 @@ public function run(): int
             if (++$loops > $this->loopMax) {
                 return 0;
             }
-            $lambda->processNextEvent($this->handler);
+
+            /**
+             * In case the execution failed, we force starting a new process regardless
+             * of $this->loopMax. This is because an uncaught exception could have
+             * left the application in a non-clean state.
+             */
+            if (!$lambda->processNextEvent($this->handler)) {
+                return 1;
```

Comment:
I think bref chose to return zero in the same scenario?

Reply:
Yes, Bref will always exit the application with code 0. Im not sure why. I'll research if there is a difference from AWS's point of view

Reply:
Hm.. I didnt find anything interesting... Changed to exit with 0

Reply:
I guess it's because the daemon process didn't fail. It just decided to stop due to a subtask failing?